craft_providers/base.py

```diff
@@ -786,6 +792,63 @@ def _post_setup_network(self, executor: Executor) -> None:
         """
         return
 
+    def _mount_shared_cache_dirs(self, executor: Executor) -> None:
+        """Mount a shared cache directories for this base.
+
+        e.g.
+            pip cache (normally $HOME/.cache/pip)
+
+        This will only be run if caching is enabled for this instance.
+
+        This step should usually be extended, but may be overridden if common
+        cache directories are in unusual locations.
+        """
+        if self._cache_path is None:
+            logger.debug("No cache path set, not mounting cache directories.")
+            return
+
+        # Get the real path with additional tags.
+        host_base_cache_path = pathlib.Path(
+            os.path.realpath(self._cache_path), self.compatibility_tag, str(self.alias)
```

Comment:
why os.path.realpath?

Reply:
To avoid someone pass a relative path that cannot find when mounting.